src/sage/combinat/root_system/type_hyperbolic.py

```diff
@@ -0,0 +1,724 @@
+                [3, 3, 1, 3, 2, 2, 2, 2, 2],
+                [2, 2, 3, 1, 3, 2, 2, 2, 2],
+                [2, 2, 2, 3, 1, 2, 3, 2, 2],
+                [2, 2, 2, 2, 2, 1, 3, 2, 2],
+                [2, 2, 2, 2, 3, 3, 1, 3, 2],
+                [2, 2, 2, 2, 2, 2, 3, 1, 3],
+                [2, 2, 2, 2, 2, 2, 2, 3, 1]
+            ]),
+    (144, 1, 8): CoxeterMatrix([
+                [1, 3, 2, 2, 2, 2, 2, 2, 2],
+                [3, 1, 3, 3, 2, 2, 2, 2, 2],
+                [2, 3, 1, 2, 3, 2, 2, 2, 2],
+                [2, 3, 2, 1, 2, 3, 2, 2, 2],
+                [2, 2, 3, 2, 1, 2, 3, 2, 2],
+                [2, 2, 2, 3, 2, 1, 2, 3, 2],
+                [2, 2, 2, 2, 3, 2, 1, 2, 3],
+                [2, 2, 2, 2, 2, 3, 2, 1, 3],
+                [2, 2, 2, 2, 2, 2, 3, 3, 1]
+            ]),
+    (144, 1, 9): CoxeterMatrix([
+                [1, 3, 2, 2, 2, 2, 2, 2, 2, 2],
+                [3, 1, 2, 3, 2, 2, 2, 2, 2, 2],
+                [2, 2, 1, 3, 2, 2, 2, 2, 2, 2],
+                [2, 3, 3, 1, 3, 2, 2, 2, 2, 2],
+                [2, 2, 2, 3, 1, 3, 2, 2, 2, 2],
+                [2, 2, 2, 2, 3, 1, 3, 2, 2, 2],
+                [2, 2, 2, 2, 2, 3, 1, 3, 2, 2],
+                [2, 2, 2, 2, 2, 2, 3, 1, 3, 2],
+                [2, 2, 2, 2, 2, 2, 2, 3, 1, 3],
+                [2, 2, 2, 2, 2, 2, 2, 2, 3, 1]
+            ]),
+    (144, 1, 10):CoxeterMatrix([
+                [1, 3, 2, 2, 2, 2, 2, 2, 2, 2],
+                [3, 1, 2, 3, 2, 2, 2, 2, 2, 2],
+                [2, 2, 1, 3, 2, 2, 2, 2, 2, 2],
+                [2, 3, 3, 1, 3, 2, 2, 2, 2, 2],
+                [2, 2, 2, 3, 1, 3, 2, 2, 2, 2],
+                [2, 2, 2, 2, 3, 1, 3, 2, 2, 2],
+                [2, 2, 2, 2, 2, 3, 1, 3, 2, 2],
+                [2, 2, 2, 2, 2, 2, 3, 1, 3, 2],
+                [2, 2, 2, 2, 2, 2, 2, 3, 1, 4],
+                [2, 2, 2, 2, 2, 2, 2, 2, 4, 1]
+            ]),
+    (144, 1, 11): CoxeterMatrix([
+                [1, 2, 3, 2, 2, 2, 2, 2, 2],
+                [2, 1, 3, 2, 2, 2, 2, 2, 2],
+                [3, 3, 1, 3, 2, 2, 2, 2, 2],
+                [2, 2, 3, 1, 3, 2, 2, 2, 2],
+                [2, 2, 2, 3, 1, 2, 3, 2, 2],
+                [2, 2, 2, 2, 2, 1, 3, 2, 2],
+                [2, 2, 2, 2, 3, 3, 1, 3, 2],
+                [2, 2, 2, 2, 2, 2, 3, 1, 3],
+                [2, 2, 2, 2, 2, 2, 2, 3, 1]
+            ]),
+}
+
+
+class CoxeterType_Hyperbolic(CoxeterType):
+    r"""
+    Coxeter type hyperbolic.
+
+    """
+
+    def __init__(self, accronym, position):
+        """
+        EXAMPLES::
+
+            sage: C = CoxeterType(["Hyp", (142, 1, 3)])
+            sage: C
+            Coxeter type with Humphrey's datum (142, 1, 3)
+
+        TESTS::
+
+            sage: TestSuite(e).run()
+        """
+        self._acronym = accronym
+        self._position = tuple(position)
+
+        if position not in hyperbolic_coxeter_matrices:
+            raise ValueError(f"Position {position} is not a valid hyperbolic Coxeter type position")
```

Comment:
```suggestion
            raise ValueError(f"position {position} is not a valid hyperbolic Coxeter type position")
```
We follow Python’s conventions.